Maximum result modal width

Invertisment opened the pull request:
This caps the witdh so that the modal would be fully visible on narrow screens:

![image](https://user-images.githubusercontent.com/1641263/183351432-ea972924-4a3f-4293-95f1-bc770a192498.png)

Invertisment committed:
Show results without modal

Invertisment committed:
Basic pages

Invertisment committed:
Fix unneeded horizontal scroll

Invertisment committed:
Results: Fix x-overflow

Invertisment committed:
Editor doesn't force parens

Invertisment commented:
I'm closing this PR because my fork contains code which has more fixes than this PR talks about.
You can access it here:
https://invertisment.github.io/4ever-clojure/

Invertisment committed:
Fix modal size

borkdude commented:
@oxalorg How about we move this project to its own Github organisation and we have more active maintainers?

We could host it under 4ever-clojure.github.io for example along with https://github.com/oxalorg/4clojure-solutions-archive (which I forgot to include in the currently deployed github pages version).

Invertisment commented:
Also I decided to not show test statuses in the modal and I refactored it to show them in the original test list. The modal is only shown when all tests pass.

oxalorg commented:
@Invertisment Great work on removing the modals and other fixes, I tried them out on your hosted version 🎉 ! I apologize for not responding on this PR sooner.

If you would not mind, I would really like to re-open this PR and merge the changes in, super appreciate your contributions! 🙌 🙏

oxalorg commented:
@borkdude I have been thinking about the same thing, let me get to the open PRs first and get them sorted this week and I'll ping you soon! Thanks for bearing with me 🙏

Invertisment commented:
Could you also look at my current fork? Should I make a PR after?
Also I changed how github actions work because it didn't work with your config.

oxalorg commented:
Thanks I've reopened the PR, so all the latest commits on main from your fork are reflected in this PR now.

Regarding the changes to GH Actions, can you explain what was going wrong because the  existing action works fine here: https://oxalorg.github.io/4ever-clojure/

Invertisment commented:
> what was going wrong

I wanted to set up my own version and it couldn't succeed. This log shows that it can't set up pages: 
https://github.com/Invertisment/4ever-clojure/actions/runs/3980187951/jobs/6823118057

So instead I simply replaced the flow with what worked from a different project.
My current deployment compiles the `public` into a branch called `gh-pages` and in settings I deploy from there.

borkdude commented:
I don't think that gh-actions change is necessary. When I implemented this I first came across the plugin that @Invertisment is using but then discovered that there is an official github plugin for this, which is currently used.

borkdude commented:
Sorry, I didn't read the full message and didn't see the error log that @Invertisment posted. It's strange that it didn't work for you, it did work in my fork as well:

https://borkdude.github.io/4ever-clojure

borkdude commented:
I see https://github.com/actions/deploy-pages is at v1.2.3 now, perhaps try that one?

Invertisment committed:
Revert to original Github Pages config

Invertisment commented:
I took the original and when I chose the `beta` option in settings then the page build worked.
I didn't need to try the new version, the current one worked. I think their API was down or I had the wrong option in settings.

I reverted the pages buildscript and now it worked: https://github.com/Invertisment/4ever-clojure/actions/runs/3986713029/jobs/6835611610

borkdude commented:
Awesome, maybe we can document this for people who make PRs so we can preview them

Invertisment commented:
I'll add a comment into the workflow file. This way people will find it when they'll have problems.

borkdude commented:
Also feel free to upgrade the action to v1.2.3

Invertisment committed:
Info on how to run own instance via GitHub Actions

Invertisment commented:
I didn't update but I added comments.

borkdude commented:
LGTM

Invertisment committed:
GH actions yaml: add 'if' to doc

borkdude commented:
@oxalorg 

Shall I go ahead and merge this?
 
Are you OK with me creating the 4ever clojure org where I add you and me as an admin (and possibly other contributors, e.g. @Invertisment if he wants) and have a fully automated deployment over there?

Invertisment commented:
Don't merge this. I found bad paren behavior on mobile Firefox and others.

If I type `inc 1)` and type the missing paren then it doesn't recover and every browser I tested actually works differently.
I tested Mobile Firefox, Desktop Firefox and mobile Chromium. :/ 

The original behavior at least recovers from the `inc 1)` but it's still bad. So both configs are bad.

Invertisment committed:
Extended and basic editor mode +Settings modal

Invertisment commented:
I added a settings button. But I don't know how to prevent reevaluation of the form when I request the `code` to change. 
I found that I add some text and change the setting then the input disappears. So I request the code value but it changes the error message.

Also I don't know why but code auto-formatting doesn't work for me at all. In neither version. But I don't know why it is this way as this was broken before.

borkdude commented:
I think leaving the advanced option off by default is a good choice.

Invertisment committed:
Move settings modal into its own file

Invertisment commented on src/app/problem.cljs:
```diff
@@ -16,114 +19,180 @@
-        [:span test]
+       [render-result-item
+        test
         (if passed?
```
This check is bad because on error (let's say we call `([inc] 1 2 3 4 5)` the custom eval returns a string:

```
(defn eval-string [source]
  (try (sci/eval-string* context source)
       (catch :default e
         (with-out-str (error-handler source e)))))
```
This is what it does in the current version.

Repro:
Go to https://4clojure.oxal.org/#/problem/38 

Enter `[]` into the field
Outcome: pass fail pass

borkdude replied:
OK, let's make a separate issue for this

Invertisment replied:
I did: https://github.com/oxalorg/4ever-clojure/issues/78

Invertisment commented:
I was trying to make the error message appear on top of the result when using restricted functions but for some reason there are so many eval functions that I'd have to refactor all of them to make it work.

Also if there is an error in eval-string it returns a string, not something proper. So either it's a boolean or a string if it's an error.

borkdude commented:
@Invertisment Agreed. I think it would be best to cut these changes into multiple smaller PRs accompanied by an issue describing the problem, maybe? Then it will be easier to review and merge them.

Invertisment commented:
I deleted the commit about restrictions and now only settings commit and modal changes are here.

In any case -- I don't understand what it means to get a like from the repo owner but no merge or instructions whether there needs to be a change or anything else.

borkdude commented:
@Invertisment 

Have you deployed this at https://invertisment.github.io/4ever-clojure/? 

What are the things I should exactly test to see differences with the currently deployed version?

If @oxalorg doesn't object, I'll merge this after reviewing those changes in your deployed version.

But I'd be more comfortable if this project lived under an organization with automated deployment as I would still have to ask @oxalorg to deploy these changes after I would merge this. Let me know what you think @oxalorg.

Invertisment commented:
It auto-redeployed after my force push: https://github.com/Invertisment/4ever-clojure/actions.

- I added the "Editor settings" modal which adds persistent setting of the editor extension mode (the checkbox is persisted) (closes https://github.com/oxalorg/4ever-clojure/issues/65 and https://github.com/oxalorg/4ever-clojure/issues/64). This setting disables matching paren deletion and insertion.
- I changed the padding of the top-page message about the solutions because it was making the page scroll horizontally (the original has slight horizontal scroll)
- I changed the modal size and padding so that it would not overflow on mobiles (narrow screens) (closes https://github.com/oxalorg/4ever-clojure/issues/61)
- I changed the result display from display in a modal only to display in the original screen. This has a problem when the test is long and I added `overflow-y: hidden`. I didn't address this UI bug on narrow screens (basically the `<dot>uh-oh` message will hide the test code and I disabled the scroll (this could probably be mitigated by hiding `uh-oh` message when responsive CSS's step function detects a narrow screen (there is a jump if you resize))).
- I renamed `modal-open` variable to `success-modal-open` and changed the modal to only be used in success condition. It also shows the modal only when all tests pass.

borkdude commented:
LGTM. If @oxalorg hasn't responded within 24 hours, I'll hit the (squash) merge button

oxalorg committed:
Merge branch 'main' into main

oxalorg commented:
@Invertisment went through the PR LGTM! Thank you for this AMAZING work and bearing through with me. The changes are merged and live :) 

![](https://media0.giphy.com/media/tIeCLkB8geYtW/giphy.gif?cid=ecf05e47a8zxip1mx6la8awkebkfhc1xlenulvb4mphm6ku0&rid=giphy.gif&ct=g)

I've made you a collaborator of this repository! 😊

Invertisment commented:
> I've made you a collaborator of this repository!

Does this mean that I can commit directly? But should I commit directly?

borkdude commented:
@Invertisment 

I think it's better if at least one of us reviews a PR before merging, but I would say low-risk things like doc changes can be merged directly.

Do you agree @oxalorg ?

@oxalorg Thanks for setting up the auto-deploy to your https://4clojure.oxal.org domain and adding @Invertisment as a collaborator!

borkdude commented:
@Invertisment Side note after the fact:

In this playground (also SCI) I'm using nextjournal/clojure mode from NPM directly (without a build, see the HTML)

https://babashka.org/scittle/codemirror.html

It seems I still get paren balancing but not the annoying limitation of not being able to delete something. Perhaps this is an even better default than what's currently there?

Invertisment commented:
Yeah, this one seems to be really good.
If it would work this way then there wouldn't be a need for the basic mode and whole settings modal.

borkdude commented:
Well I guess we could just use that from NPM directly:

https://github.com/nextjournal/lang-clojure

and drop the CLJS dependency on nextjournal/clojure-mode.

Invertisment commented:
Do we want to keep the settings modal though?

borkdude commented:
We could get rid of it for now, until we have settings again :)
